File: lib/components/scan_button.dart
import 'package:flutter/material.dart';
import 'package:simple_barcode_scanner/simple_barcode_scanner.dart';
import '../api.dart';
import './global_variables.dart';
import './alert.dart';

class ScanButton extends StatefulWidget {
  const ScanButton({Key? key}) : super(key: key);

  @override
  _ScanButtonState createState() => _ScanButtonState();
}

class _ScanButtonState extends State<ScanButton> {
  String barCodeScanResult = '';
  List<String> foundThings = [];

  @override
  Widget build(BuildContext context) {
    return Padding(
      padding: const EdgeInsets.all(16.0),
      child: ElevatedButton(
        onPressed: () async {
          if (lookingForThings.isEmpty) {
            showAlert(context, 'Nothing Selected',
                'You need to select something to filter for');
            return;
          }

          showProcessingDialog(context, "Scanning your item...");
          print("Scanning dialog shown");

          try {
            final res = await Navigator.push(
              context,
              MaterialPageRoute(
                  builder: (context) => const SimpleBarcodeScannerPage()),
            );

            if (res is String) {
              setState(() => barCodeScanResult = res);
              await makeGetRequest(barCodeScanResult, foundThings, context);
            } else {
              Navigator.of(context).pop();
              throw Exception('Barcode scanning failed or was cancelled');
            }
          } catch (e) {
            print('Error occurred: $e');
            if (mounted) {
              Navigator.of(context).pop();
              showAlert(context, 'Error',
                  'An error occurred while processing your request. Please try again.');
            }
          }
        },
        style: ElevatedButton.styleFrom(
          backgroundColor: Colors.blue,
          foregroundColor: Colors.white,
        ),
        child: const Text('SCAN', style: TextStyle(color: Colors.white)),
      ),
    );
  }
}

File: composeApp/src/commonMain/kotlin/org/whatsinmyfood/project/components/scan_button.kt
/*import android.app.AlertDialog
import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.util.Log
import android.view.View
import android.widget.Button
import androidx.activity.result.contract.ActivityResultContracts
import androidx.appcompat.app.AppCompatActivity
import androidx.fragment.app.FragmentActivity
import kotlin.Exception

class ScanButtonActivity : AppCompatActivity() {

    private var barCodeScanResult: String = ""
    private val foundThings: MutableList<String> = mutableListOf()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_scan_button)

        val scanButton = findViewById<Button>(R.id.scan_button)

        scanButton.setOnClickListener {
            if (GlobalVariables.lookingForThings.isEmpty()) {
                showAlert(this, "Nothing Selected", "You need to select something to filter for")
                return@setOnClickListener
            }

            showProcessingDialog(this, "Scanning your item...")
            Log.d("ScanButtonActivity", "Scanning dialog shown")

            try {
                barcodeScannerLauncher.launch(Intent(this, SimpleBarcodeScannerActivity::class.java))
            } catch (e: Exception) {
                Log.e("ScanButtonActivity", "Error occurred: $e")
                showAlert(this, "Error", "An error occurred while processing your request. Please try again.")
            }
        }
    }

    // ActivityResultLauncher for handling barcode scanning result
    private val barcodeScannerLauncher = registerForActivityResult(ActivityResultContracts.StartActivityForResult()) { result ->
        if (result.resultCode == RESULT_OK) {
            val res = result.data?.getStringExtra("SCAN_RESULT") // Replace with actual key if needed
            res?.let {
                barCodeScanResult = it
                makeGetRequest(barCodeScanResult, foundThings, this)
            } ?: run {
                showAlert(this, "Error", "Barcode scanning failed or was cancelled")
            }
        } else {
            showAlert(this, "Error", "Barcode scanning failed or was cancelled")
        }
    }

    private fun showAlert(context: Context, title: String, message: String) {
        AlertDialog.Builder(context)
            .setTitle(title)
            .setMessage(message)
            .setPositiveButton("OK", null)
            .create()
            .show()
    }

    private fun showProcessingDialog(context: Context, message: String) {
        AlertDialog.Builder(context)
            .setMessage(message)
            .setCancelable(false)
            .create()
            .show()
    }
}
*/